install/install-base.php

```diff
@@ -4,131 +4,169 @@
 	 *
 	 * @since 1.7
+	 * @since 3.8 Static method.
 	 *
 	 * @param bool $networkwide Whether the plugin is activated for all sites in the network or just the current site.
 	 * @return void
 	 */
-	public function activate( $networkwide ) {
-		$this->do_for_all_blogs( 'activate', $networkwide );
+	public static function activate( $networkwide ) {
+		static::do_for_all_blogs( 'activate', $networkwide );
 	}
 
 	/**
-	 * Plugin activation
+	 * Plugin activation.
 	 *
 	 * @since 0.5
+	 * @since 3.8 Static method.
 	 *
 	 * @return void
 	 */
-	protected function _activate() {
-		// Can be overridden in child class
+	protected static function _activate() {
+		// Can be overridden in child class.
 	}
 
 	/**
 	 * Plugin deactivation for multisite.
 	 *
 	 * @since 0.1
+	 * @since 3.8 Static method.
 	 *
 	 * @param bool $networkwide Whether the plugin is deactivated for all sites in the network or just the current site.
 	 * @return void
 	 */
-	public function deactivate( $networkwide ) {
-		$this->do_for_all_blogs( 'deactivate', $networkwide );
+	public static function deactivate( $networkwide ) {
+		static::do_for_all_blogs( 'deactivate', $networkwide );
 	}
 
 	/**
-	 * Plugin deactivation
+	 * Plugin deactivation.
 	 *
 	 * @since 0.5
+	 * @since 3.8 Static method.
 	 *
 	 * @return void
 	 */
-	protected function _deactivate() {
-		// Can be overridden in child class
+	protected static function _deactivate() {
+		// Can be overridden in child class.
 	}
 
 	/**
-	 * Site creation on multisite ( to set default options )
+	 * Site creation on multisite (to set default options).
 	 *
 	 * @since 2.6.8
+	 * @since 3.8 Static method.
 	 *
 	 * @param WP_Site $new_site New site object.
 	 * @return void
 	 */
-	public function new_site( $new_site ) {
+	public static function new_site( $new_site ) {
 		switch_to_blog( $new_site->id );
-		$this->_activate();
+		static::_activate();
 		restore_current_blog();
 	}
+
+	/**
+	 * Activation or deactivation for all blogs.
+	 *
+	 * @since 1.2
+	 * @since 3.8 Static method.
+	 *
+	 * @param string $what        Either 'activate' or 'deactivate'.
+	 * @param bool   $networkwide Whether the plugin is (de)activated for all sites in the network or just the current site.
+	 * @return void
+	 */
+	protected static function do_for_all_blogs( $what, $networkwide ) {
+		if ( is_multisite() && $networkwide ) {
+			// Network.
+			foreach ( self::get_sites() as $blog_id ) {
+				switch_to_blog( $blog_id );
+				'activate' === $what ? static::_activate() : static::_deactivate();
+			}
+			restore_current_blog();
+		} else {
+			// Single blog.
+			'activate' === $what ? static::_activate() : static::_deactivate();
+		}
+	}
+
+	/**
+	 * Returns (and cache) the list of sites on multi-site.
+	 *
+	 * @since 3.8
+	 *
+	 * @return array
+	 */
+	protected static function get_sites(): array {
+		global $wpdb;
+
+		if ( empty( self::$sites ) ) {
+			self::$sites = $wpdb->get_col( "SELECT blog_id FROM $wpdb->blogs" );
+		}
```

Comment:
Instead of having our own method, could we now use the function `get_sites()`. It seems to me that `number` set to 0 would retrurn all sites (I did not test). 
We would benefit of the WP cache.